### PR TITLE
Keep try syntax in async exception rewriting

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/74362

The async exception handler rewriter (which is only involved for an `await` inside a `catch` or `finally`, such as awaiting the disposal in an `await foreach`) produced a bound node for the `try` whose syntax was not recognized by the `SyntaxBindingUtilities.BindsToTryStatement` assertion.
In the problematic scenario, the syntax was the block of the accessor containing the local function with `await foreach`.
The fix is to synthesize the nodes for this `try` with more specific syntax, such as the `await foreach` statement.

The `AsyncInCatchFilterLifted` test was affected and so I updated `IsClosureScope`

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs [hunk 1]
```diff
@@ -58,11 +58,6 @@ private static void VerifyParams(IParameterSymbol parameter, bool isParamArray =
             VerifyParams(parameter.GetSymbol<ParameterSymbol>(), isParamArray: isParamArray, isParamCollection: isParamCollection);
         }
 
-        private static string ExpectedOutput(string output)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs:61]: >ExpectedOutput [](http://example.com/codeflow?start=30&length=14)

I do not think moving this method to a shared location with this name an implementation is appropriate. The name is too general/unspecific and the logic is not applicable in every test.

File: src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs [hunk 2]
```diff
@@ -2625,5 +2625,10 @@ public interface IAsyncDisposable
     }
 }
 ";
+
+        internal static string ExpectedOutput(string output)
```
Comment (AlekseyTs) [inline src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs:2629]: >ExpectedOutput [](http://example.com/codeflow?start=31&length=14)

The name doesn't reflect the purpose of the method, therefore, I do not think it should be used for a global/shared helper.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

**Review by 333fred — APPROVED** [review]

**Pull request merged**